packages/core/src/store/pin-store.ts

```diff
@@ -14,8 +14,8 @@ export class PinStore {
       readonly resolve: (path: string) => Promise<CID>
     ) {}
 
-    async open(): Promise<void> {
-        await this.stateStore.open()
+    async open(networkName: string): Promise<void> {
+        await this.stateStore.open(networkName)
```

Comment:
So the "StateStore" holds both the state of any document commits/tips loaded and the tips of any documents created. Does that all happen through the PinStore?

Here it seems there could be other entities besides the PinStore that could use the same stateStore, however the `open` method presumably should be called only once so it seems out of place here.

If my understanding is correct my suggestion would be to open the stateStore in the Ceramic initialization instead.

Reply:
hmm, the `StateStore` *is* sometimes used directly, but always via a handle from the pinStore.  I can't imagine there ever being a time when we would have a `StateStore` but not have a PinStore.  In fact, with the stuff Sergey is working on now, I expect all these internal pieces to become even more hidden away behind the new `DocStateRepository` interface.  So I think we want to be moving in the direction of pushing more of these pieces down behind interfaces rather than pulling more of them up to be used directly in higher layers

Reply:
Ok so state store is only used if you explicitly choose to pin something? If that is the case then it makes sense how it is

Reply:
Yeah, there will only be state in the state store for docs that are pinned

Reply:
🙏 thanks for clarity